cordapp-example/workflows-kotlin/src/test/kotlin/com/example/test/flow/CreateMoneyFlowTests.kt

```diff
@@ -0,0 +1,100 @@
+package com.example.test.flow
+
+import com.example.flow.CreateMoneyFlow
+import com.example.state.CashState
+import net.corda.core.flows.FlowLogic
+import net.corda.core.identity.CordaX500Name
+import net.corda.core.utilities.contextLogger
+import net.corda.core.utilities.getOrThrow
+import net.corda.testing.core.singleIdentity
+import net.corda.testing.node.MockNetwork
+import net.corda.testing.node.MockNetworkParameters
+import net.corda.testing.node.StartedMockNode
+import net.corda.testing.node.TestCordapp
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+
+class CreateMoneyFlowTests {
+    private lateinit var network: MockNetwork
+    private lateinit var bank: StartedMockNode
+    private lateinit var owner: StartedMockNode
+    private var cashValue: Long = 30
+
+    private fun validFlow() = CreateMoneyFlow.Initiator(cashValue, owner.info.singleIdentity())
+    private fun validFuture() = bank.startFlow(validFlow())
+
+    @Before
+    fun setup() {
+        network = MockNetwork(MockNetworkParameters(cordappsForAllNodes = listOf(
+                TestCordapp.findCordapp("com.example.contract"),
+                TestCordapp.findCordapp("com.example.flow")
+        )))
+
+        val x500Name = CordaX500Name.parse("O=Bank,L=Paris,C=FR")
+        bank = network.createPartyNode(legalName = x500Name)
+        owner = network.createPartyNode()
+
+        // For real nodes this happens automatically, but we have to manually register the flow for tests.
+        owner.registerInitiatedFlow(CreateMoneyFlow.Acceptor::class.java)
+
+        network.runNetwork()
+    }
+
+    @After
+    fun tearDown() {
+        network.stopNodes()
+    }
+
+    @Test
+    fun `throw when creator is not a bank`() {
+        val flow = CreateMoneyFlow.Initiator(cashValue, bank.info.singleIdentity())
```

Comment:
`bank` should start the `CreateMoneyFlow`; not `owner` 👇

Reply:
I'm actually allowing for anyone to be cash creator but on `PaybackFlow.Acceptor` I'm checking if creator is a `bank`